src/styles/main.scss

```diff
@@ -1,3 +1,51 @@
-body {
+* {
   margin: 0;
+  padding: 0;
+  box-sizing: border-box;
+}
+
+.box-container {
+  display: grid;
+
+  @media (min-width: 300px) {
+    grid-template-columns: minmax(300px, 1fr);
+    grid-template-rows: repeat(6, 300px);
+  }
+
+  @media (min-width: 600px) {
+    grid-template-columns: repeat(2, minmax(300px, 1fr));
+    grid-template-rows: repeat(3, 300px);
+  }
+
+  @media (min-width: 900px) {
+    grid-template-columns: repeat(3, minmax(300px, 1fr));
+    grid-template-rows: repeat(2, 300px);
+  }
```

Comment:
`grid-template-columns: repeat(auto-fill, minmax(300px, 1fr));`
try use this, maybe you could cut amount of media